v-next/core/test/plugins/plugin-validation.ts

```diff
@@ -0,0 +1,115 @@
+import assert from "node:assert/strict";
+import { describe, it } from "node:test";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+
+import { validatePluginNpmDependencies } from "../../src/internal/plugins/plugin-validation.js";
+import { HardhatPlugin } from "../../src/types/plugins.js";
+
+describe("Plugins - plugin validation", () => {
+  const plugin: HardhatPlugin = {
+    id: "example-plugin",
+    npmPackage: "example",
+  };
+
+  it("should skip validation if the plugin is not from an npm package", async () => {
+    await assert.doesNotReject(async () =>
+      validatePluginNpmDependencies({
+        ...plugin,
+        npmPackage: undefined,
+      }),
+    );
+  });
+
+  describe("when the plugin has no peer deps", () => {
+    it("should pass validation if the npm package has been installed", async () => {
+      const installedPackageProjectFixture = import.meta.resolve(
+        "./fixture-projects/installed-package",
+      );
+
+      await assert.doesNotReject(async () =>
```

Comment:
The Node docs actually recommend against using `doesNotReject` / `doesNotThrow`:

> Using assert.doesNotReject() is actually not useful because there is little benefit in catching a rejection and then rejecting it again. Instead, consider adding a comment next to the specific code path that should not reject and keep error messages as expressive as possible.

Reply:
That is interesting. I was trying to find a pattern that allowed me to assert that an async function does not throw. What do we think the pattern should be in this case? Just a direct await on the function under test?

I think I prefer `assert.doesNotReject(...)` as its more explicit, but happy to hear if there are better patterns.

Reply:
In the `hardhat-utils`, I think I left the code that shouldn't throw at the top level of the test with a comment. The message is pretty similar, to be honest. But I get your point about being explicit. Additionally, using the assertion could be better if we provide a custom `message` along with the assertion.

Reply:
I am going to merge. For the moment, I have a mild preference for the explicit assertion style, but lets revisit if we find a better pattern or come to the conclusion we shouldn't be using it after some more experience.